apis/core/v1beta1/featureflag_types.go

```diff
@@ -0,0 +1,85 @@
+/*
```

Comment:
Regarding validation on flag schema 

alpha version uses a validation webhook [1] [2] and I think this is useful even if we are now handling flag configurations with `Flag` type. 

was there any reason to avoid the validation webhook ? Otherwise let's add that if there are no downsides to it. 

[1] - https://github.com/open-feature/open-feature-operator/blob/main/main.go#L281-L284
[2] - https://github.com/open-feature/open-feature-operator/blob/main/webhooks/featureflagconfiguration_webhook.go

Reply:
the validation webhook used to validate, that we have a valid json schema in the CRD, as in v1alpha1, the field was of type `string`, which means the user can basically put anything into this field. Currently we have a struct https://github.com/open-feature/open-feature-operator/pull/535/files#diff-c29b273426465350caa0db6a338509afcf85b7400bccfffd4bec81465c2e8c41R40 which already has a kubebuilder validations set (see the kubebuidlder comments). Therefore I assume this is even stricter validation than the previous one with a separate validation webhook

Reply:
Thanks for the note on validation annotations. 

Regarding the usage of structure, how flexible will this be ? For example, if we include a new field or make an exisiting field optional (ex:- https://github.com/open-feature/flagd-schemas/issues/80 where there were discussions on requirement of the variant). If we define validations at operator side, we will need a new new version of the CRD. So wouldn't it be better to stick to schema validation even if that's primitive compared to stricter rules 🤔  ?

@toddbaert @thisthat appreciate discussion on this

Reply:
We can always perform changes in `beta` versions, even breaking changes. Firstly in stable version breaking changes are not allowed.

In the API definition I went from the latest alpha version `v1alpha2.FeatureFlagConfiguration`, as that is the logical order to take here. 

To the flexibility, we can change basically anything (what I do not expect), but for example adding new fields is not considered as a breaking change and I do not see a problem here. When we perform a change (for example adding a new field), we will create a new `v1beta2` version, which will become the new hub version and potentially also create conversion webhooks (if needed). But this is a standard process in k8s. I would personally prefer a struct to a plain string in the definition

Reply:
@odubajDT yup, agree on all the points. I am also fine with updating the crd version when there's a substantial flag definition change. 

@toddbaert  do you see any other aspect on this ? Otherwise let's move ahead with this change.

Reply:
No, I agree. With the proper struct our validation doesn't add anything.

Some day we might add validation for rule logic, but we don't have that yet.